kobuki_dock_drive/src/dock_drive_states.cpp

```diff
@@ -157,8 +210,12 @@ namespace kobuki {
       // turn right, CW until get right signal from left sensor
       if(left & (DockStationIRState::FAR_RIGHT + DockStationIRState::NEAR_RIGHT)) {
         next_state = RobotDockingState::GET_STREAM;
-        next_vx = 0.5;
-        next_wz = 0.0;
+        next_vx = 0.05;
+
+      	// slight hack: smaller dock_detector values indicate robot is farther away from dock so
+      	// needs greater angle correction, should probably test around some more to find what
+        // function would work best for this
+        next_wz = 8.0*(1.0 - (double)dock_detector/160.0);
```

Comment:
what's with the 8*(1-dock_detector/160) business?  What values does dock_detector take?